Beams/MRML/vtkMRMLRTIonBeamNode.cxx

```diff
@@ -0,0 +1,514 @@
+  this->SetCollimatorAngle(node->GetCollimatorAngle());
+  this->SetCouchAngle(node->GetCouchAngle());
+
+  this->DisableModifiedEventOff();
+  this->InvokePendingModifiedEvent();
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLRTIonBeamNode::SetScene(vtkMRMLScene* scene)
+{
+  Superclass::SetScene(scene);
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLRTIonBeamNode::PrintSelf(ostream& os, vtkIndent indent)
+{
+  Superclass::PrintSelf(os,indent);
+
+  os << indent << " VSADx:   " << this->VSADx << "\n";
+  os << indent << " VSADy:   " << this->VSADy << "\n";
+  os << indent << " IsocenterToJawsDistanceX:   " << this->IsocenterToJawsDistanceX << "\n";
+  os << indent << " IsocenterToJawsDistanceY:   " << this->IsocenterToJawsDistanceY << "\n";
+  os << indent << " IsocenterToMultiLeafCollimatorDistance:   " << this->IsocenterToMultiLeafCollimatorDistance << "\n";
+  os << indent << " IsocenterToRangeShifterDistance:   " << this->IsocenterToRangeShifterDistance << "\n";
+  os << indent << " ScanningSpotSizeX:   " << this->ScanningSpotSize[0] << "\n";
+  os << indent << " ScanningSpotSizeY:   " << this->ScanningSpotSize[1] << "\n";
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLRTIonBeamNode::SetAndObserveScanSpotTableNode(vtkMRMLTableNode* node)
+{
+  if (node && this->Scene != node->GetScene())
+  {
+    vtkErrorMacro("Cannot set reference: the referenced and referencing node are not in the same scene");
+    return;
+  }
+
+  this->SetNodeReferenceID( SCANSPOT_REFERENCE_ROLE, (node ? node->GetID() : nullptr));
+
+  this->InvokeCustomModifiedEvent(Superclass::BeamGeometryModified);
+}
+
+//----------------------------------------------------------------------------
+vtkMRMLTableNode* vtkMRMLRTIonBeamNode::GetScanSpotTableNode()
+{
+  return vtkMRMLTableNode::SafeDownCast( this->GetNodeReference(SCANSPOT_REFERENCE_ROLE) );
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLRTIonBeamNode::CreateDefaultDisplayNodes()
+{
+  // Create default model display node
+  Superclass::CreateDefaultDisplayNodes();
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLRTIonBeamNode::CreateDefaultTransformNode()
+{
+  Superclass::CreateDefaultTransformNode();
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLRTIonBeamNode::CreateNewBeamTransformNode()
+{
+  // Create transform node for ion beam
+  Superclass::CreateNewBeamTransformNode();
+}
+
+//----------------------------------------------------------------------------
+void vtkMRMLRTIonBeamNode::SetVSAD( double xComponent, double yComponent)
+{
+  this->VSADx = xComponent;
+  this->VSADy = yComponent;
+  this->Modified();
+  this->InvokeCustomModifiedEvent(vtkMRMLRTBeamNode::BeamGeometryModified);

```

Comment:
In some places you use Superclass, in others the class name itself. Please make it consistent.

Reply:
Explicit class name